2.5 Release notes - Host registration improvements (#1819)
theforeman/theforeman.org: _includes/manuals/nightly/1.2_release_notes.md

### Headline features

> #### Host registration improvements

Reviewer: Due to the manual branching this change should be moved to `_includes/manuals/2.5/1.2_release_notes.md`

Packages for Foreman and its various plug-ins that are built for CentOS 8 are now being tested on CentOS 8 Stream as well.

#### Host registration improvements

> The host registration form has been redesigned and has new fields. The `Install Packages` field can be used to install additional packages while the `Repository & GPG keys` fields can be used to configure additional software repositories for yum/dnf or apt.

Reviewer 2: I didn't check, but are `Repository` and `GPG Keys` separate fields? If so, I'd be explicit:
```suggestion
The host registration form has been redesigned and has new fields. The `Install Packages` field can be used to install additional packages while the `Repository` and `GPG keys` fields can be used to configure additional software repositories for yum/dnf or apt.
```

Packages for Foreman and its various plug-ins that are built for CentOS 8 are now being tested on CentOS 8 Stream as well.

> #### Host registration improvements

Reviewer 2: I'd be consistent with the rest and add an empty line
```suggestion
#### Host registration improvements

```

> Redesigned form to generate the registration command with new fields `Install Packages` & `Repository & GPG keys`. New template kind `host_init_config` for host initial configuration (used to be `Linux registration default` template). This template is automatically assigned to every operating system. And added new API endpoint `/api/registration_commands` for generating registration command.

Reviewer 2: It can be even more verbose
```suggestion
The host registration form has been redesigned and has new fields. `Install Packages` can be used to install additional packages while `Repository & GPG keys` can be used to configure additional software repositories for yum/dnf or apt.

A new template kind `host_init_config` for host initial configuration has been added to replace the old `Linux registration default` template. This template is automatically assigned to every operating system.

Additionally, a new API endpoint `/api/registration_commands` makes it easy to generate a registration command.
```

Reviewer 3: I agree, the more verbose the better :) 

Is `Install Packages` a new field? 

If so i'd suggest "The `Install Packages` field can be used" and the same for the GPG. 

I would also say "A new `host_init_config` template replaces the old `Linux registration default` template. This new template is automatically assigned to every operating system. 

In general I am +1 to these suggestions!

Puppet 5 is officially End of Life. Foreman {{page.version}} still supports Puppet 5, but in a future release this will be removed.

> ### Host registration improvements

Reviewer 2: We don't add a section for each feature. Rather, this should be under `Headline features` (like Ubuntu 20.04 and CentOS 8 Stream).

> * Redesigned form for generating registration command with new fields `Install Packages` & `Repository & GPG keys`

Reviewer 2: ```suggestion
* Redesigned form to generate the registration command with additional `Install Packages` & `Repository & GPG keys` fields
```
